### PR TITLE
add tolerations for nodeplugin and server pods

### LINKED ISSUE #732: [RFE] Allow adding tolerations to bypass taints on master nodes
My current setup is 3 masters + 2 workers where the masters have a NoSchedule taint (as they don't serve as workers).

It will be nice if we can add tolerations for the StratefulSet (for example on Replica3 we got server-gluster-replica3-0 , server-gluster-replica3-1 & server-gluster-replica3-2) and the respectful pods (server-gluster-replica3-0-0, server-gluster-replica3-1-0 ,etc) so Kadalu can run on the master-only nodes

### PR DESCRIPTION
**Pull request by leelavg**
ref: #302
fixes: #732

Signed-off-by: Leela Venkaiah G <leelavg@thoughtexpo.com>

### CODE REVIEW

**Comment by leelavg** [comment]
- not tested yet, will change state once tested

**Comment by leelavg** [comment]
- Tested as below
- Deploy
``` yaml
---
apiVersion: kadalu-operator.storage/v1alpha1
kind: KadaluStorage
metadata:
  name: sample-pool
spec:
  type: Replica1
  storage:
    - node: worker-a
      device: /dev/loop0
    - node: worker-b
      device: /dev/loop0
    - node: worker-c
      device: /dev/loop0
```
- Add tolerations in CR and that'll be applied for server and nodeplugin pods to be scheduled on master as well
``` yaml
---
apiVersion: kadalu-operator.storage/v1alpha1
kind: KadaluStorage
metadata:
  name: sample-pool
spec:
  tolerations:
    - key: node-role.kubernetes.io/master
      operator: Exists
      effect: NoSchedule
  type: Replica1
  storage:
    - node: master-0
      device: /dev/loop0
    - node: master-1
      device: /dev/loop0
    - node: master-2
      device: /dev/loop0
    - node: worker-a
      device: /dev/loop0
    - node: worker-b
      device: /dev/loop0
    - node: worker-c
      device: /dev/loop0
```

**Review by vatsa287 — APPROVED** [review]

File: helm/kadalu/charts/operator/templates/resources.yaml [hunk 1]
```diff
@@ -42,6 +43,23 @@ spec:
...
+                      effect:
+                        type: string
+                      key:
+                        type: string
+                      operator:
+                        type: string
+                      tolerationSeconds:
+                        format: int64
+                        type: integer
+                      value:
```
Comment (vatsa287) [inline helm/kadalu/charts/operator/templates/resources.yaml:61]: Is the error/warning when "operator": "exists" and value is given, will this be handled internally by k8's?

Comment (leelavg) [reply helm/kadalu/charts/operator/templates/resources.yaml:61]: 1. Schema is validated while applying CR as usual 
2. Toleration array is indirectly validated by k8s when patching the resource, it fails logging an error

Comment (vatsa287) [reply helm/kadalu/charts/operator/templates/resources.yaml:61]: Ack.

**Review by aravindavk — APPROVED** [review]

**Review by vatsa287 — COMMENTED** [review]
I think this PR has missed running `make gen-manifests` to update the manifest files. 
I'll update with PR related to adding annotation path.

**Pull request merged**